### PR TITLE
docs: Update docs with RETENTION_MS config for stream/table

### PR DESCRIPTION
**Pull request by bvarghese1**
### Description 
Docs for RETENTION_MS config in the WITH clause while creating streams/tables

Related PR - https://github.com/confluentinc/ksql/pull/9223

### CODE REVIEW

File: docs/developer-guide/ksqldb-reference/create-table-as-select.md [hunk 1]
```diff
@@ -197,6 +197,33 @@ is used.
...
+If `RETENTION_MS` isn't set, the retention of the input stream is
+used.
+
+In join queries, the `RETENTION_MS` value is taken from the left-most stream or
+table.
+
+You can't change the retention on an existing windowed table. To change the
+retention, you must drop the stream and create it again.
+
+Note, this setting is only accepted while creating windowed tables.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-table-as-select.md:213]:
```suggestion
This setting is only accepted while creating windowed tables.

```

File: docs/developer-guide/ksqldb-reference/create-table-as-select.md [hunk 2]
```diff
@@ -197,6 +197,33 @@ is used.
...
+Additionally, the largest of `RETENTION_MS` and `RETENTION` is used while
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-table-as-select.md:214]:
```suggestion
Additionally, the larger of `RETENTION_MS` and `RETENTION` is used while

```

File: docs/developer-guide/ksqldb-reference/create-table-as-select.md [hunk 3]
```diff
@@ -197,6 +197,33 @@ is used.
...
+creating the backing topic if it doesn't exist.
+
+For example, to retain the computed windowed aggregation results for a week,
+you might run the following query(retention_ms = 7 days and retention = 2 days):
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/create-table-as-select.md:218]:
```suggestion
you might run the following query, where `retention_ms` = 7 days and `retention` = 2 days:

```

**Review by JimGalasyn — APPROVED** [review]
LGTM, with a few suggestions.

**Comment by bvarghese1** [comment]
> LGTM, with a few suggestions.

Made the changes as suggested.

**Pull request merged**